java/daikon/dcomp/DCInstrument24.java

```diff
@@ -0,0 +1,4916 @@
+            li.add(ce);
+          }
+        }
+        inst_index++;
+      }
+    } catch (DynCompError e) {
+      throw e;
+    } catch (Throwable t) {
+      throw new DynCompError(
+          String.format(
+              "Unexpected error processing %s.%s.%n", mgen.getClassName(), mgen.getName()),
+          t);
+    }
+  }
+
+  /**
+   * Create a worklist item.
+   *
+   * @param target label where to start operand stack simulation
+   * @param stack state of operand stack at target
+   */
+  protected static void addLabelToWorklist(Label target, OperandStack24 stack) {
+    OperandStack24 existing = worklistHistory.get(target);
+    if (existing == null) {
+      if (debugOperandStack) {
+        System.out.println(
+            "push to worklist: " + target + ", " + labelIndexMap.get(target) + ", stack: " + stack);
+      }
+      worklistHistory.put(target, stack.getClone());
+      worklist.add(new WorkItem(labelIndexMap.get(target), stack.getClone()));
+    } else {
+      // will throw if stacks don't match
+      verifyOperandStackMatches(target, existing, stack);
+      // stacks match, don't add duplicate to worklist
+      if (debugOperandStack) {
+        System.out.println(
+            "duplicate worklist item: "
+                + target
+                + ", "
+                + labelIndexMap.get(target)
+                + ", stack: "
+                + stack);
+      }
+    }
+  }
+
+  /**
+   * Verify that the operand stacks match at a label.
+   *
+   * @param target label where control flow merges
+   * @param existing state of operand stack at target
+   * @param current state of operand stack at transfer to target
+   */
+  protected static void verifyOperandStackMatches(
+      Label target, OperandStack24 existing, OperandStack24 current) {
+    if (existing.equals(current)) {
+      if (debugOperandStack) {
+        System.out.println("operand stacks match at: " + target);
+      }
+      return;
+    }
+    // stacks don't match
+    if (debugOperandStack) {
+      System.err.flush();
+      System.out.flush();
+      System.out.println("operand stacks do not match at label:" + target);
+      System.out.println("existing stack: " + existing);
+      System.out.println("current stack: " + current);
+      System.out.flush();
+    }
+    throw new DynCompError("operand stacks do not match at label:" + target);
+  }
+
+  /**
+   * Adds the method name and containing class name to {@code skip_methods}, the list of
+   * uninstrumented methods.
+   *
+   * @param method method to add to skipped_methods list
+   */
+  void skip_method(String method) {
+    skipped_methods.add(method);
+  }
+
+  /**
+   * Returns the list of uninstrumented methods. (Note: instrument_jdk() needs to have been called
+   * first.)
+   */
+  public List<String> get_skipped_methods() {
+    return new ArrayList<>(skipped_methods);
+  }
+
+  /**
+   * Adds a try/catch block around the entire method. If an exception occurs, the tag stack is
```

Comment:
"If an exception occurs": does that mean in this method, or at run time for the instrumented code?  I think the latter, but please clarify.